src_features/signMessage/ui_common_signMessage.c

```diff
@@ -1,39 +1,29 @@
 #include "os_io_seproxyhal.h"
+#include "lib_standard_app/crypto_helpers.h"
 #include "common_ui.h"
 
 unsigned int io_seproxyhal_touch_signMessage_ok(void) {
-    uint8_t privateKeyData[INT256_LENGTH];
-    uint8_t signature[100];
-    cx_ecfp_private_key_t privateKey;
     uint32_t tx = 0;
-    io_seproxyhal_io_heartbeat();
-    os_perso_derive_node_bip32(CX_CURVE_256K1,
-                               tmpCtx.messageSigningContext.bip32.path,
-                               tmpCtx.messageSigningContext.bip32.length,
-                               privateKeyData,
-                               NULL);
-    io_seproxyhal_io_heartbeat();
-    cx_ecfp_init_private_key(CX_CURVE_256K1, privateKeyData, 32, &privateKey);
-    explicit_bzero(privateKeyData, sizeof(privateKeyData));
     unsigned int info = 0;
-    io_seproxyhal_io_heartbeat();
-    cx_ecdsa_sign(&privateKey,
-                  CX_RND_RFC6979 | CX_LAST,
-                  CX_SHA256,
-                  tmpCtx.messageSigningContext.hash,
-                  sizeof(tmpCtx.messageSigningContext.hash),
-                  signature,
-                  sizeof(signature),
-                  &info);
-    explicit_bzero(&privateKey, sizeof(privateKey));
+    if (bip32_derive_ecdsa_sign_rs_hash_256(CX_CURVE_256K1,
+                                            tmpCtx.messageSigningContext.bip32.path,
+                                            tmpCtx.messageSigningContext.bip32.length,
+                                            CX_RND_RFC6979 | CX_LAST,
+                                            CX_SHA256,
+                                            tmpCtx.messageSigningContext.hash,
+                                            sizeof(tmpCtx.messageSigningContext.hash),
+                                            G_io_apdu_buffer + 1,
+                                            G_io_apdu_buffer + 1 + 32,
+                                            &info) != CX_OK) {
+        THROW(0x6F00);
```

Comment:
defined error code ?

Reply:
Not yet.
I have added `APDU_RESPONE_UNKNOWN`.